pkg/mobile/types_test.go

```diff
@@ -0,0 +1,59 @@
+package mobile
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestMetricValidate(t *testing.T) {
+
+	testCases := []struct {
+		Name           string
+		Metric         Metric
+		Valid          bool
+		ExpectedReason string
+	}{
+		{
+			Name:           "Empty metric should be invalid",
+			Metric:         Metric{},
+			Valid:          false,
+			ExpectedReason: "missing clientId in payload",
```

Comment:
shouldn't the message be `empty metric is invalid`?

Reply:
The validate function executes checks in sequence where the first check is the clientId. For that reason this was the behaviour I intended. Happy to change it it but I think it's good

Reply:
I was just going by the name of the test which mentions `Empty metrics`. I guess `missing  clientId` is still correct in that case.